spec/index.bs

```diff
@@ -1514,169 +1518,160 @@ interface GPUMapMode {
-        1. |this|.{{GPUBuffer/[[state]]}} must be [=buffer state/unmapped=]
-        1. |mode| must contain exactly one of {{GPUMapMode/READ}} or {{GPUMapMode/WRITE}}.
-        1. If |mode| contains {{GPUMapMode/READ}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_READ}}.
-        1. If |mode| contains {{GPUMapMode/WRITE}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_WRITE}}.
-
-        Issue: Do we validate that |mode| contains only valid flags?
-    </div>
-</div>
-
-### <dfn method for=GPUBuffer>getMappedRange(offset, size)</dfn> ### {#GPUBuffer-getMappedRange}
+<dl dfn-type=method dfn-for=GPUBuffer>
+    : <dfn>mapAsync(mode, offset, size)</dfn>
+    ::
 
-<div algorithm="GPUBuffer.getMappedRange">
-    <strong>|this|:</strong> of type {{GPUBuffer}}.
+        <div algorithm=GPUBuffer.mapAsync>
+            **Called on:** {{GPUBuffer}} |this|.
 
-    **Arguments:**
+            **Arguments:**
+            <pre class=argumentdef for="GPUBuffer/mapAsync(mode, offset, size)">
+                |mode|:
+                |offset|:
+                |size|:
+            </pre>
 
-    - {{GPUSize64}} |offset|
-    - {{GPUSize64}} |size|
+            **Returns:** Promise<void>
 
-    **Returns:** {{ArrayBuffer}}
+            Issue(gpuweb/gpuweb#605): Handle error buffers once we have a description of the error monad.
 
-    1. If |size| is unspecified:
+            - If |size| is unspecified:
+                - If |offset| > |this|.{{GPUBuffer/[[size]]}}, reject with an {{OperationError}} and stop.
+                - Let |rangeSize| be |this|.{{GPUBuffer/[[size]]}} - |offset|.
 
-        - If |offset| &ge; this.{{[[size]]}}, throw an {{OperationError}} and stop.
-        - Let |rangeSize| be this.{{[[size]]}} - |offset|.
+                Otherwise, let |rangeSize| be |size|.
 
-        Otherwise, let |rangeSize| be |size|.
+            - Let |invalid| be `true` if any of the following conditions are unsatisfied.
+                <div class=validusage>
+                    - |this| is a [=valid=] {{GPUBuffer}}.
+                    - |offset| is a multiple of 4.
+                    - |rangeSize| is a multiple of 4.
+                    - |offset| + |rangeSize| is less or equal to |this|.{{[[size]]}}
+                    - |this|.{{GPUBuffer/[[state]]}} is [=buffer state/unmapped=]
+                    - |mode| contains exactly one of {{GPUMapMode/READ}} or {{GPUMapMode/WRITE}}.
+                    - If |mode| contains {{GPUMapMode/READ}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_READ}}.
+                    - If |mode| contains {{GPUMapMode/WRITE}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_WRITE}}.
+
+                    Issue: Do we validate that |mode| contains only valid flags?
+                </div>
 
-    1. If this call doesn't follow the [$getMappedRange Valid Usage$]:
+            - If |invalid| is `true`:
```

Comment:
Perhaps(?) these two steps can be combined into one like the if-then above, like,

```
- If any of the following conditions are unsatisfied:

    <div class=validusage>
        ...
    </div>

    then:

    - Record a validation error ...
    - Return a promise rejected ...

Reply:
That reads pretty nicely when render, IMO. I definitely prefer the phrasing. Thanks for the suggestion!